pynest/nest/tests/test_NodeCollection.py

```diff
@@ -593,15 +593,17 @@ def test_apply_positions(self):
         param = nest.spatial.distance
         # Single target position
         target = [[1., 2.], ]
-        for source in n:
+        for i in range(len(n)):
+            source = n[i]
```

Comment:
I am not sure I agree that this is the best way to do it. It seems to me that it would be very unintuitive for the user. How much slower would it be to also return the metadata so that we could do it the old way?